src/app/static/src/app/store/load/actions.ts

```diff
@@ -42,15 +44,17 @@ export const actions: ActionTree<LoadState, RootState> & LoadActions = {
         const files = objectContents.files;
         const savedState = objectContents.state;
 
-        await api<LoadActionTypes, LoadErrorActionTypes>(context)
-            .withSuccess("UpdatingState")
-            .withError("LoadFailed")
-            .postAndReturn<Dict<LocalSessionFile>>("/session/files/", files)
-            .then(() => {
-                if (state.loadingState == LoadingState.UpdatingState) {
-                    dispatch("updateStoreState", savedState);
-                }
-            });
+        await getFilesAndLoad(context, files, savedState)
+    },
+
+    async loadFromSnapshot(context, snapshotContents) {
+        const {commit} = context;
+        commit({type: "SettingFiles", payload: null});
+
+        router.push("/", () => {
```

Comment:
We currently need to be on the root route by then time we reload the page or get a 404 from the backend. As discussed, it would be good at some point to make the backend recognise the /versions route so user can reload page from the versions screen without error.